tests/src/test/scala/cats/tests/BinCodecInvariantMonoidalSuite.scala

```diff
@@ -8,7 +8,7 @@ import cats.implicits._
 import cats.Eq
 import cats.kernel.laws.discipline.{MonoidTests, SemigroupTests}
 import org.scalacheck.{Arbitrary, Gen}
-import kernel.compat.lazyList._
+import kernel.compat.scalaVersionSpecific._
```

Comment:
Is it really necessary to have an import like this in user code? Can't we hide the compat layer somehow?

Reply:
The compat layer is hidden (package private to `cats`).  It's used here because the test referenced `Stream`, and is cross-compiled against 2.12 and 2.13, so the tests itself need the same cross scala compat machinery provided by the `scalaVersionSpecific` 
Users who are not cross compiling do not use any of these compat to use cats. But users who are might need their own compat layer if they are using a deprecated type from 2.12. E.g. if a cross-compiling user is using `Stream`, there is simply no `Stream` instances in Cats for 2.13, only `Lazy List` instances. The user will need the same compat layer (could be copied from Cats) to use `Stream` in 2.12 and `LazyList` in 2.13.